### PR TITLE
Fix emulator

### PR DESCRIPTION
**Pull request by kochol**
I add emulator code and tune it a little and add WebSocket Server to it so now the webcore can connect to emulator with websocket.

### CODE REVIEW

File: Makefile.include [hunk 1]
```diff
@@ -15,7 +15,7 @@ OBJDUMP  := objdump
 AR       := ar
 AS       := as
 
-OPTFLAGS ?= -O3
```
Comment (hadideveloper) [inline Makefile.include:18]: Due to STM32 memory limit, we need this optimization level, so we may remove optimization for emulator only

**Review by hadideveloper — CHANGES_REQUESTED** [review]
Makefile.include should be changed

File: Makefile.include (hunk 1 above)
Comment (kochol) [reply Makefile.include:18]: If you look at the line 10 of this file you will see this options are only for EMULATOR builds.

The STM32 build options are at line 33

Comment (hadideveloper) [reply Makefile.include:18]: Ah, OK

**Review by hadideveloper — APPROVED** [review]

**Pull request merged**